crates/mun_language_server/src/state/protocol/dispatcher.rs

```diff
@@ -41,6 +41,36 @@ impl<'a> RequestDispatcher<'a> {
         Ok(self)
     }
 
+    /// Try to dispatch the event as the given Request type on the thread pool.
+    pub fn on<R>(
+        &mut self,
+        f: fn(LanguageServerSnapshot, R::Params) -> anyhow::Result<R::Result>,
```

Comment:
What kind of function is `f`? Maybe use that as a parameter name?

Reply:
Renamed to `compute_response_fn`, also in different places.